src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyWorkerService.java

```diff
@@ -0,0 +1,936 @@
+
+                if (factor.isVariableSet()) {
+                    monitoredBranchDetailResult.setP(sensitivityValue.getFunctionReference());
+                }
+
+                if (!mapEncounteredBranches.get(functionId)) {  // <branch, contingency> not already handled
+                    // compute the variation for the monitored branch
+                    SensitivitiesByBranch sensitivitiesByBranch = sensitivitiesByBranches.get(functionId);
+                    Map<String, Double> generatorsSensitivities = sensitivitiesByBranch.getSensitivitiesByContingency().get(contingencyId);
+                    Map<String, GeneratorCapping> generatorCappings = new HashMap<>();
+                    double variation = computeVariationForMonitoredBranch(network, nonEvacuatedEnergyInputs, factor, monitoredBranchThreshold, generatorsSensitivities, sensitivityValue, generatorCappings, monitoredBranchDetailResult);
+
+                    // update monitored branch result with the cappings information
+                    Map<EnergySource, Double> cappingsByEnergySource = new EnumMap<>(EnergySource.class);
+                    for (Map.Entry<String, GeneratorCapping> entry : generatorCappings.entrySet()) {
+                        Generator generator = network.getGenerator(entry.getKey());
+                        GeneratorCapping capping = !monitoredBranchDetailResult.getGeneratorsCapping().containsKey(entry.getKey())
+                            ? entry.getValue() : monitoredBranchDetailResult.getGeneratorsCapping().get(entry.getKey());
+                        capping.setCumulatedCapping(capping.getCumulatedCapping() + entry.getValue().getCapping());
+                        monitoredBranchDetailResult.getGeneratorsCapping().put(entry.getKey(), capping);
+                        cappingsByEnergySource.putIfAbsent(generator.getEnergySource(), 0.);
+                        cappingsByEnergySource.put(generator.getEnergySource(), cappingsByEnergySource.get(generator.getEnergySource()) + capping.getCumulatedCapping());
+                    }
+                    monitoredBranchDetailResult.getCappingByEnergySource().putAll(cappingsByEnergySource);
+                    monitoredBranchDetailResult.setOverallCapping(cappingsByEnergySource.values().stream().mapToDouble(d -> d).sum());
+
+                    // keeping only the max computed variation
+                    if (!Double.isNaN(variation) && maxVariationForMonitoredBranch < variation) {
+                        maxVariationForMonitoredBranch = variation;
+                        maxGeneratorsCappings = new HashMap<>(generatorCappings.entrySet()
+                            .stream().collect(toMap(Map.Entry::getKey, e -> e.getValue().getCapping())));
+                    }
+
+                    mapEncounteredBranches.put(functionId, Boolean.TRUE);
+                }
+            }
+        }
+
+        if (maxVariationForMonitoredBranch < 0.3) {
```

Comment:
Add constant at the top of the file like Epsilon?

Reply:
Done